<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>ExploreJuncTable</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IRE1_SARS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ExploreJuncTable</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>ire1_sars/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunire1sarstreec4d98f23e59063b5159e400cfdda8585138b90f0targetblankc4d98f2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/ire1-sars/tree/c4d98f23e59063b5159e400cfdda8585138b90f0" target="_blank">c4d98f2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunire1sarstreec4d98f23e59063b5159e400cfdda8585138b90f0targetblankc4d98f2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/ire1-sars/tree/c4d98f23e59063b5159e400cfdda8585138b90f0" target="_blank">c4d98f2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/.Snakefile.swp
    Ignored:    code/._.DS_Store
    Ignored:    code/.snakemake/
    Ignored:    code/Ref/
    Ignored:    code/bams/
    Ignored:    code/idxstats/
    Ignored:    code/juncs/
    Ignored:    code/logs/

Untracked files:
    Untracked:  output/Galaxy36-Control-Sars2-1.annotated.bed.gz
    Untracked:  output/Galaxy39-Control-Sars2-2.annotated.bed.gz
    Untracked:  output/Galaxy42-Control-Sars2-3.annotated.bed.gz
    Untracked:  output/Galaxy54-IRE1KO-Sars2-1.annotated.bed.gz
    Untracked:  output/Galaxy57-IRE1KO-Sars2-2.annotated.bed.gz
    Untracked:  output/Galaxy60-IRE1KO-Sars2-3.annotated.bed.gz
    Untracked:  output/TotalReadsPerChrom.tsv

Unstaged changes:
    Modified:   analysis/ExploreJuncTable.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/ExploreJuncTable.Rmd</code>) and HTML (<code>docs/ExploreJuncTable.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/ire1-sars/blob/c4d98f23e59063b5159e400cfdda8585138b90f0/analysis/ExploreJuncTable.Rmd" target="_blank">c4d98f2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-03-29
</td>
<td>
initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/ire1-sars/c4d98f23e59063b5159e400cfdda8585138b90f0/docs/ExploreJuncTable.html" target="_blank">c4d98f2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-03-29
</td>
<td>
initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Letita and Long are interested in the hyptothesis that SARS-Cov2 genome may have decoy Ire1 binding/splice sites to out-compete Ire-1-dependent non-canonical splicing of Xbp1 which is involved in innate immune response signalling. They have infected WT or IRE1-KO cells with SARS-Cov2, and performed stranded RNA-seq, and are interested in exploring potential Ire-1 dependent splice sites, particularly on SAR-Cov2 RNA. I have tabulated their splice junction alignments from their bam files with <code>regtools juntions extract</code> and <code>regtools junctions annotate</code> using the rules described the <a href="../code/Snakefile">snakefile</a>. Here I will quickly explore the results before presenting them to Letita and Long.</p>
<p>first let’s load some R libraries</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✓ ggplot2 3.3.5     ✓ purrr   0.3.4
✓ tibble  2.1.3     ✓ dplyr   0.8.3
✓ tidyr   1.1.0     ✓ stringr 1.4.0
✓ readr   1.3.1     ✓ forcats 0.4.0</code></pre>
<pre><code>── Conflicts ───────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<p>Now let’s read in the tabulated junction files. You can download the repository yourself and get these junction files:</p>
<p><code>git clone git@github.com:bfairkun/ire1-sars.git</code>,</p>
<p>then change to the <code>analysis</code> directory, and try running this code:</p>
<pre class="r"><code>junc_files &lt;- list.files(&quot;../output&quot;, pattern=&quot;*.bed.gz&quot;, full.names = T) %&gt;%
  setNames(nm=str_replace(., &quot;.+?-(.+?-Sars2-.+?)\\.annotated.bed.gz&quot;, &quot;\\1&quot;))
junc_files</code></pre>
<pre><code>                                      Control-Sars2-1 
&quot;../output/Galaxy36-Control-Sars2-1.annotated.bed.gz&quot; 
                                      Control-Sars2-2 
&quot;../output/Galaxy39-Control-Sars2-2.annotated.bed.gz&quot; 
                                      Control-Sars2-3 
&quot;../output/Galaxy42-Control-Sars2-3.annotated.bed.gz&quot; 
                                       IRE1KO-Sars2-1 
 &quot;../output/Galaxy54-IRE1KO-Sars2-1.annotated.bed.gz&quot; 
                                       IRE1KO-Sars2-2 
 &quot;../output/Galaxy57-IRE1KO-Sars2-2.annotated.bed.gz&quot; 
                                       IRE1KO-Sars2-3 
 &quot;../output/Galaxy60-IRE1KO-Sars2-3.annotated.bed.gz&quot; </code></pre>
<pre class="r"><code>dat &lt;- lapply(junc_files, read_tsv) %&gt;% 
   bind_rows(.id=&quot;sample&quot;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  chrom = col_character(),
  start = col_double(),
  end = col_double(),
  name = col_character(),
  score = col_double(),
  strand = col_character(),
  splice_site = col_character(),
  acceptors_skipped = col_double(),
  exons_skipped = col_double(),
  donors_skipped = col_double(),
  anchor = col_character(),
  known_donor = col_double(),
  known_acceptor = col_double(),
  known_junction = col_double(),
  gene_names = col_character(),
  gene_ids = col_character(),
  transcripts = col_character()
)
Parsed with column specification:
cols(
  chrom = col_character(),
  start = col_double(),
  end = col_double(),
  name = col_character(),
  score = col_double(),
  strand = col_character(),
  splice_site = col_character(),
  acceptors_skipped = col_double(),
  exons_skipped = col_double(),
  donors_skipped = col_double(),
  anchor = col_character(),
  known_donor = col_double(),
  known_acceptor = col_double(),
  known_junction = col_double(),
  gene_names = col_character(),
  gene_ids = col_character(),
  transcripts = col_character()
)
Parsed with column specification:
cols(
  chrom = col_character(),
  start = col_double(),
  end = col_double(),
  name = col_character(),
  score = col_double(),
  strand = col_character(),
  splice_site = col_character(),
  acceptors_skipped = col_double(),
  exons_skipped = col_double(),
  donors_skipped = col_double(),
  anchor = col_character(),
  known_donor = col_double(),
  known_acceptor = col_double(),
  known_junction = col_double(),
  gene_names = col_character(),
  gene_ids = col_character(),
  transcripts = col_character()
)
Parsed with column specification:
cols(
  chrom = col_character(),
  start = col_double(),
  end = col_double(),
  name = col_character(),
  score = col_double(),
  strand = col_character(),
  splice_site = col_character(),
  acceptors_skipped = col_double(),
  exons_skipped = col_double(),
  donors_skipped = col_double(),
  anchor = col_character(),
  known_donor = col_double(),
  known_acceptor = col_double(),
  known_junction = col_double(),
  gene_names = col_character(),
  gene_ids = col_character(),
  transcripts = col_character()
)
Parsed with column specification:
cols(
  chrom = col_character(),
  start = col_double(),
  end = col_double(),
  name = col_character(),
  score = col_double(),
  strand = col_character(),
  splice_site = col_character(),
  acceptors_skipped = col_double(),
  exons_skipped = col_double(),
  donors_skipped = col_double(),
  anchor = col_character(),
  known_donor = col_double(),
  known_acceptor = col_double(),
  known_junction = col_double(),
  gene_names = col_character(),
  gene_ids = col_character(),
  transcripts = col_character()
)
Parsed with column specification:
cols(
  chrom = col_character(),
  start = col_double(),
  end = col_double(),
  name = col_character(),
  score = col_double(),
  strand = col_character(),
  splice_site = col_character(),
  acceptors_skipped = col_double(),
  exons_skipped = col_double(),
  donors_skipped = col_double(),
  anchor = col_character(),
  known_donor = col_double(),
  known_acceptor = col_double(),
  known_junction = col_double(),
  gene_names = col_character(),
  gene_ids = col_character(),
  transcripts = col_character()
)</code></pre>
<pre class="r"><code>head(dat)</code></pre>
<pre><code># A tibble: 6 x 18
  sample chrom start   end name   score strand splice_site acceptors_skipp…
  &lt;chr&gt;  &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;                  &lt;dbl&gt;
1 Contr… MN90…    85 15794 JUNC…    411 +      GT-AG                      0
2 Contr… MN90…    65 21552 JUNC… 106394 +      CT-AA                      0
3 Contr… MN90…    65 25381 JUNC… 147114 +      CT-AC                      0
4 Contr… MN90…    83 26251 JUNC…  20035 +      CT-AC                      0
5 Contr… MN90…    64 26468 JUNC… 150525 +      TC-GG                      0
6 Contr… MN90…    69 27041 JUNC…  40613 +      AC-TC                      0
# … with 9 more variables: exons_skipped &lt;dbl&gt;, donors_skipped &lt;dbl&gt;,
#   anchor &lt;chr&gt;, known_donor &lt;dbl&gt;, known_acceptor &lt;dbl&gt;,
#   known_junction &lt;dbl&gt;, gene_names &lt;chr&gt;, gene_ids &lt;chr&gt;,
#   transcripts &lt;chr&gt;</code></pre>
<p>Now let’s explore the data a bit…</p>
<p>Sort of similar to normalizing data to the total number of mapped reads, let’s plot the total number of spliced reads and consider normalizing later results to a factor based on this.</p>
<pre class="r"><code>dat %&gt;%
  group_by(sample) %&gt;%
  summarise(TotalJuncs = sum(score)) %&gt;%
  ggplot(aes(x=sample, y=TotalJuncs)) +
  geom_col() +
  theme_bw()</code></pre>
<p><img src="figure/ExploreJuncTable.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ire1-sars/blob/c4d98f23e59063b5159e400cfdda8585138b90f0/docs/figure/ExploreJuncTable.Rmd/unnamed-chunk-3-1.png" target="_blank">c4d98f2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-03-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so there are more total junc reads in the control samples than the IRE1KO. This could just be unequal sequencing depth, or biological differences in amount of IRE-dependent splicing consistent with our hypothesis. How about let’s replot that but just filter for spliceosomal (GT-AG) introns.</p>
<pre class="r"><code>dat %&gt;%
  filter(splice_site == &quot;GT-AG&quot;) %&gt;%
  group_by(sample) %&gt;%
  summarise(TotalJuncs = sum(score)) %&gt;%
  ggplot(aes(x=sample, y=TotalJuncs)) +
  geom_col() +
  theme_bw()</code></pre>
<p><img src="figure/ExploreJuncTable.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ire1-sars/blob/c4d98f23e59063b5159e400cfdda8585138b90f0/docs/figure/ExploreJuncTable.Rmd/unnamed-chunk-4-1.png" target="_blank">c4d98f2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-03-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Wow, now the opposite effect. The control samples have less spliceosome dependent splicing, so the previous plot isn’t due to just differences in total read depth. This is consistent with their hypothesis.</p>
<p>Another quick quality check is that Long mentioned that &gt;90% of reads map to SARS cov2 contig. Let’s verify that by counting junctions mapping to chromosome versus SARS cov2 contig.</p>
<pre class="r"><code>dat %&gt;%
  mutate(HostOrVirus = case_when(
    chrom == &quot;MN908947.3&quot; ~ &quot;SARS-Cov2&quot;,
    TRUE ~ &quot;Host&quot;
  )) %&gt;%
  group_by(sample, HostOrVirus) %&gt;%
  summarise(TotalJuncs = sum(score)) %&gt;%
  ggplot(aes(x=sample, y=TotalJuncs, fill=HostOrVirus)) +
  geom_col(position=&quot;fill&quot;) +
  theme_bw() +
  ylab(&quot;Fraction junc reads&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/ExploreJuncTable.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ire1-sars/blob/c4d98f23e59063b5159e400cfdda8585138b90f0/docs/figure/ExploreJuncTable.Rmd/unnamed-chunk-5-1.png" target="_blank">c4d98f2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-03-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok so IRE-KO has a much larger fraction of reads originating from SARS, consistent with impaired ability to fight off infection. This important to keep in mind for downstream analyses.</p>
<p>Let’s verify IRE1-dependent non-canonical splicing of a 26nt intron in xbp1.</p>
<pre class="r"><code>dat %&gt;%
  filter(gene_names==&quot;XBP1&quot;) %&gt;%
  mutate(len = end-start) %&gt;%
  filter( len &lt; 30)</code></pre>
<pre><code># A tibble: 2 x 19
  sample chrom  start    end name  score strand splice_site
  &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;      
1 Contr… chr22 2.92e7 2.92e7 JUNC…     2 -      CT-CA      
2 Contr… chr22 2.92e7 2.92e7 JUNC…     8 -      CT-CA      
# … with 11 more variables: acceptors_skipped &lt;dbl&gt;, exons_skipped &lt;dbl&gt;,
#   donors_skipped &lt;dbl&gt;, anchor &lt;chr&gt;, known_donor &lt;dbl&gt;,
#   known_acceptor &lt;dbl&gt;, known_junction &lt;dbl&gt;, gene_names &lt;chr&gt;,
#   gene_ids &lt;chr&gt;, transcripts &lt;chr&gt;, len &lt;dbl&gt;</code></pre>
<p>Ok, at least in these two control samples, but not in any of the IRE-KO samples I can see some splice junctions of a 26nt non-canonical intron. Cool.</p>
<p>Now let’s check fraction junc reads derived from non-spliceosomal introns (as determined by GT-AG or AT-AC, or GC-AG motifs at intron boundaries).</p>
<pre class="r"><code>dat %&gt;%
  count(splice_site) %&gt;%
  arrange(desc(n))</code></pre>
<pre><code># A tibble: 257 x 2
   splice_site       n
   &lt;chr&gt;         &lt;int&gt;
 1 GT-AG       1058938
 2 CT-AC         68400
 3 GC-AG         15753
 4 AA-TT         12739
 5 TT-AA         11214
 6 AA-CA         11116
 7 TT-TT          9783
 8 AA-TA          8981
 9 AT-TT          8653
10 AA-AA          8459
# … with 247 more rows</code></pre>
<pre class="r"><code>dat %&gt;%
  mutate(HostOrVirus = case_when(
    chrom == &quot;MN908947.3&quot; ~ &quot;SARS-Cov2&quot;,
    TRUE ~ &quot;Host&quot;
  )) %&gt;%
  mutate(SpliceSiteType = case_when(
    splice_site %in% c(&quot;GT-AG&quot;, &quot;AT-AC&quot;, &quot;GC-AG&quot;) ~ &quot;spliceosomal&quot;,
    TRUE ~ &quot;Non-spliceosomal&quot;
  )) %&gt;%
  group_by(sample, HostOrVirus, SpliceSiteType) %&gt;%
  summarise(TotalJuncs = sum(score)) %&gt;%
  ggplot(aes(x=sample, y=TotalJuncs, fill=SpliceSiteType)) +
  geom_col(position=&quot;fill&quot;) +
  theme_bw() +
  ylab(&quot;Fraction junc reads&quot;) +
  facet_wrap(~HostOrVirus) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/ExploreJuncTable.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/ire1-sars/blob/c4d98f23e59063b5159e400cfdda8585138b90f0/docs/figure/ExploreJuncTable.Rmd/unnamed-chunk-7-1.png" target="_blank">c4d98f2</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-03-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so the splicing in the host genome is predominantly spliceosomal, as expected. In SARS-Cov2 its primarily non-spliceosomal. Which doesn’t necessarily surprise me. And also keep in mind that non-canonical splicing is indistinguishable from deletions compared to the reference genome. So some of the ‘non-canonical splicing’ may actual just be deletion variants from the SARS-Cov2 reference. What I think makes sense to look for is the number of canonical junctions in SARS-Cov2 normalized to the total number of SARS-Cov2 reads. I think we would need to do another command to get that total number of SARS-Cov2 reads per sample, perhaps using <code>samtools idxstats</code> <a href="https://www.biostars.org/p/14531/">see link</a>. I already did that in the Snakefile.</p>
<pre class="r"><code>TotalCountsPerChrom &lt;- read_tsv(&quot;../output/TotalReadsPerChrom.tsv&quot;, col_names = c(&quot;chrom&quot;, &quot;size&quot;, &quot;counts&quot;, &quot;x&quot;, &quot;FN&quot;)) %&gt;%
  mutate(sample = str_replace(FN, &quot;.+?-(.+?-Sars2-.+?)\\.txt&quot;, &quot;\\1&quot;))</code></pre>
<pre><code>Parsed with column specification:
cols(
  chrom = col_character(),
  size = col_double(),
  counts = col_double(),
  x = col_double(),
  FN = col_character()
)</code></pre>
<pre class="r"><code>TotalCountsSARS &lt;- TotalCountsPerChrom %&gt;%
  filter(chrom == &quot;MN908947.3&quot;) %&gt;%
  select(sample, TotalCounts = counts)

TotalCountsSARS</code></pre>
<pre><code># A tibble: 6 x 2
  sample          TotalCounts
  &lt;chr&gt;                 &lt;dbl&gt;
1 Control-Sars2-2   194448214
2 Control-Sars2-1   185141825
3 IRE1KO-Sars2-2    266115623
4 IRE1KO-Sars2-3    252758643
5 IRE1KO-Sars2-1    261692333
6 Control-Sars2-3   187376587</code></pre>
<p>Ok, consistent with the plot based on total splice junction counts, it is clear that the IRE1-KO samples hve more SARS Cov2 reads (though technically I did not normalize for total read depth here).</p>
<p>In any case, back to counting the total number of non-canonical splicing reads on SARS-Cov2 normalized to the total number of SARS Cov2 reads:</p>
<pre class="r"><code>dat %&gt;%
  mutate(HostOrVirus = case_when(
    chrom == &quot;MN908947.3&quot; ~ &quot;SARS-Cov2&quot;,
    TRUE ~ &quot;Host&quot;
  )) %&gt;%
  filter(HostOrVirus == &quot;SARS-Cov2&quot;) %&gt;%
  mutate(SpliceSiteType = case_when(
    splice_site %in% c(&quot;GT-AG&quot;, &quot;AT-AC&quot;, &quot;GC-AG&quot;) ~ &quot;spliceosomal&quot;,
    TRUE ~ &quot;Non-spliceosomal&quot;
  )) %&gt;%
  group_by(sample, SpliceSiteType) %&gt;%
  summarise(TotalJuncs = sum(score)) %&gt;%
  inner_join(TotalCountsSARS, by=&quot;sample&quot;) %&gt;%
  mutate(FractionJuncsOutOfTotalReads = TotalJuncs/TotalCounts*100) %&gt;%
  ggplot(aes(x=sample, y=FractionJuncsOutOfTotalReads)) +
  geom_col() +
  facet_wrap(~SpliceSiteType, scales = &quot;free&quot;) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/ExploreJuncTable.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>It’s not clear to me that there is any more non-canonical splicing of SARS-Cov2 in IRE1-KO cells than in in WT. Maybe there are more spliceosomal reads, but those only make up a small small fraction of the total junctions on SAR-Cov2 so I’m not sure how meaningful that is.</p>
<p>If I knew more about IRE-dependent splicing (eg the intron sizes, the preferred motifs) then maybe I could be better about filtering for the introns that I believe to be truly IRE-dependent, and maybe I would see a signal. I’ll leave that for Letita and Long to consider and investigate further.</p>
<p>Another question I have about their hypothesis is if the model is that SARS-Cov2 inhibits Xbp1 splicing by outcompeting for Ire1, Ire1-dependent splicing isn’t necessary for this competition and all that needs to happen is Ire1 binding to SARSCov2 RNA (though granted, splicing is something we can observe with RNA-seq, so it made sense to look for splicing).</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.4.0   stringr_1.4.0   dplyr_0.8.3     purrr_0.3.4    
[5] readr_1.3.1     tidyr_1.1.0     tibble_2.1.3    ggplot2_3.3.5  
[9] tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0 xfun_0.8         haven_2.3.1      lattice_0.20-38 
 [5] colorspace_1.4-1 vctrs_0.3.1      generics_0.0.2   htmltools_0.3.6 
 [9] yaml_2.2.0       utf8_1.1.4       rlang_0.4.10     later_0.8.0     
[13] pillar_1.4.2     withr_2.4.1      glue_1.3.1       DBI_1.1.0       
[17] dbplyr_1.4.2     modelr_0.1.8     readxl_1.3.1     lifecycle_0.1.0 
[21] munsell_0.5.0    gtable_0.3.0     workflowr_1.6.2  cellranger_1.1.0
[25] rvest_0.3.5      evaluate_0.14    labeling_0.3     knitr_1.23      
[29] httpuv_1.5.1     fansi_0.4.0      broom_0.5.2      Rcpp_1.0.5      
[33] promises_1.0.1   backports_1.1.4  scales_1.1.0     jsonlite_1.6    
[37] farver_2.1.0     fs_1.3.1         hms_0.5.3        digest_0.6.20   
[41] stringi_1.4.3    grid_3.6.1       rprojroot_2.0.2  cli_2.2.0       
[45] tools_3.6.1      magrittr_1.5     crayon_1.3.4     whisker_0.3-2   
[49] pkgconfig_2.0.2  xml2_1.3.2       reprex_0.3.0     lubridate_1.7.4 
[53] rstudioapi_0.10  assertthat_0.2.1 rmarkdown_1.13   httr_1.4.1      
[57] R6_2.4.0         nlme_3.1-140     git2r_0.26.1     compiler_3.6.1  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
